pkg/scheduler/factory.go

```diff
@@ -163,24 +160,27 @@ type Configurator struct {
 
 	PodMaxBackoffSeconds int64
 
-	enableNonPreempting bool
-
 	// framework configuration arguments.
 	Registry                     framework.Registry
 	Plugins                      *schedulerapi.Plugins
 	PluginConfig                 []schedulerapi.PluginConfig
 	PluginConfigProducerRegistry *plugins.ConfigProducerRegistry
-	nodeInfoSnapshot             *nodeinfosnapshot.Snapshot
+
+	nodeInfoSnapshot *nodeinfosnapshot.Snapshot
 
 	factoryArgs        *PluginFactoryArgs
 	configProducerArgs *plugins.ConfigProducerArgs
+
+	// Always check all predicates even if the middle of one predicate fails.
+	alwaysCheckAllPredicates bool
+
+	enableNonPreempting bool
 }
 
-// NewConfigFactory initializes the default implementation of a Configurator. To encourage eventual privatization of the struct type, we only
-// return the interface.
+// setup initializes the Configurator. It should be called before
```

Comment:
uhm... rethinking about this, `Create` calls should be calling this instead. And what this is doing is completing fields. Maybe `complete` is a better name for the function. We have similar patterns when building app configuration.

Reply:
I had the same thought but the problem is that some of the create functions call each other? So, if this method is called twice, some fields might be overwritten.  It might be enough to have the call in `CreateFromConfig` and `CreateFromProvider` but we should avoid having it in `CreateFromKeys`.  WDYT?

Reply:
leave it separate for now. But call it `complete`.

Reply:
Is the updated comment ok? Is there a better description for the method?